### ISSUE TITLE
Comment Date Block

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#30573]
Here is an initial take on the design for the Comment Date block. This block should only be available in the Site Editor, and potentially only when editing certain templates (Index, Single, Singular, Page).

![comment-date](https://user-images.githubusercontent.com/846565/113897387-b7eacc80-97c2-11eb-95f4-4c473917190e.png)

Design feedback is politely requested for all aspects:

* Icon
* Placeholder when there is no context present
* Toolbar actions
* Inspector settings
* Name
* Description

Figma link [here](https://www.figma.com/file/AagoBClTm0AgpfYCSKQl0m/Comment-blocks?node-id=501%3A1238).

**Comment by annezazu** [comment]
@jameskoster thanks so much for this exploration! While you're exploring this, I wanted to loop in some feedback that came out of a stream from a contributor where she mentioned that the date selection is currently unstyled:

<img width="446" alt="Screen Shot 2021-04-07 at 11 06 17 AM" src="https://user-images.githubusercontent.com/26996883/113906479-8a395f80-9791-11eb-8ef4-77306ce4983b.png">

Is this something you plan to address here? I wasn't able to see it in the figma file so wanted to check to be sure.

**Comment by jameskoster** [comment]
@annezazu I think that is a separate issue altogether. Looks like a bug in the Post Date block.

**Comment by annezazu** [comment]
Facepalm -- got my date blocks confused. Thank you. Sorry about that!

**Comment by paaljoachim** [comment]
Hi James

It looks like you have the added options needed for this block.
One thing. I assume it will use the same date style as the rest of the WP backend. 
As the date will change depending on the WP settings date and time settings.

**Comment by jameskoster** [comment]
> I assume it will use the same date style as the rest of the WP backend.

Yup, the date format option should be respected here. But as in the Post Date block, it should be customisable:

<img width="284" alt="Screenshot 2021-04-21 at 09 43 24" src="https://user-images.githubusercontent.com/846565/115524465-12396200-a286-11eb-9107-9a4b5d553c2d.png">

**Comment by carolinan** [comment]
The first iteration of this block as "Post Comment Date" was added in https://github.com/WordPress/gutenberg/pull/24854

**Comment by justintadlock** [comment]
Noting that a common use case for the comment published date is to wrap it in the permalink for the comment.  Maybe a "link to comment" would make for a good option.

**Comment by SantosGuillamot** [comment]
Hello! 👋   I would like to wrap up all the issues related to the Post Comments block to start working on a Comment Loop block as explained in [this tracking issue](https://github.com/WordPress/gutenberg/issues/34994). I'd like to make a quick summary to ensure we are aligned on this one:

## Description

Add the date of the specific comment.

## Potential Settings

- [x] **Typography**: It should be the same as the [Post Date Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-date).
- [x] **Color Settings**: It should be the same as the [Post Date Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-date).
- [x]  **Format Settings**: It should be the same as the [Post Date Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-date).
- [x] **Link Settings**:
    - Link to comment: Clicking the date will take the user to the comment URL.

## References

- This should be similar to the [Post Date Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-date).
- There is already a basic version of the Post Comment Date Block: [Pull Request](https://github.com/WordPress/gutenberg/pull/24854) and [GitHub code](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-date).

Of course, any feedback is welcome 🙂

**Comment by gziolo** [comment]
I see that https://github.com/WordPress/gutenberg/pull/35112 for Link Settings is close to ready. @carolinan, do you plan to continue working on other enhancements that @SantosGuillamot proposed?

**Comment by carolinan** [comment]
After #35112 there is nothing left to do on the checklist. I only mentioned that it was a partial pull request because it can't be tested inside a comment template /comment loop yet.

**Comment by gziolo** [comment]
That's awesome. Thank you for the clarification @carolinan. Let's close this issue once #35112 land then 😄 

I wasn't aware that other features require only feature toggles. As simple as that 🎉
